src/viam/examples/modules/complex/gizmo/api.cpp

```diff
@@ -22,8 +22,10 @@ GizmoRegistration::GizmoRegistration(const google::protobuf::ServiceDescriptor*
     : ResourceRegistration(service_descriptor){};
 
 std::shared_ptr<ResourceServer> GizmoRegistration::create_resource_server(
-    std::shared_ptr<ResourceManager> manager) {
-    return std::make_shared<GizmoServer>(manager);
+    std::shared_ptr<ResourceManager> manager, Server& server) {
+    auto gs = std::make_shared<GizmoServer>(std::move(manager));
+    server.register_service(gs.get());
+    return gs;
```

Comment:
Per PR description, after talking with @stuqdog, I've changed `create_resource_server` to take in a `Server&` and `register_service` with it instead of having a separate `register_server` method on every `[TypeOfResource]Server`.